### ISSUE TITLE
Analysis fails when one of its input files is already being imported

### ISSUE THREAD

**Issue by hackdna** [refinery-platform/refinery-platform#2791]
* Specific code commit: v1.6.3
* Version of the web browser and OS: any
* Environment where the error occurred: any

### Steps to reproduce
1. Import a data set with data files referenced by URLs
1. Share the data set with another user
1. Run an analysis on this data set
1. Log out
1. Sign in as the user from step 2
1. Run an analysis on the same data set using the same input files

### Observed behavior
Analysis fails

### Expected behavior
Analysis completes successfully

### Notes
* Running two import tasks on the same file is not (and should not) be allowed
* Exiting the second file import task could start analysis before the first file import task completes which would result in analysis failure
* Subsequent analyses should be able to use the state of the file import task that is already running for a given file

**Comment by jkmarx** [comment]
@hackdna Please add comments here.

**Comment by scottx611x** [comment]
We ran into this again during the release-1.6.5 testing

**Comment by scottx611x** [comment]
@hackdna Do you have any thoughts on this?

**Comment by hackdna** [comment]
Did you actually verify that running analyses using files imported more than 24 hours ago fails? If so, what's the error message? Even if it is the case then that is a different issue than "analysis fails when one of its input files is already being imported" which is certainly not critical.

**Comment by scottx611x** [comment]
Yes I've verified this is the case. Please feel free to test it out on the test site.

A basic example from an older dataset: https://test.stemcellcommons.org/data_sets/4d7f9959-80bc-4927-b918-69c16fd6a43c/#/files/
```
==> log/celeryd-w2.log <==
2018-08-29 17:02:30,682 DEBUG    file_store.tasks:151 import_file[6e119df9-ae7b-4c4c-b0c4-b1da784365f8] - Importing FileStoreItem with UUID '167cc543-99ff-4de0-b67c-bca791de594f'
2018-08-29 17:02:30,686 ERROR    file_store.tasks:170 import_file[6e119df9-ae7b-4c4c-b0c4-b1da784365f8] - File import is already in progress for 'eb/13/SRR064952.fastq.gz' - task ID: 'ab585446-3fcc-46f0-add7-6c9e060ae40c'

==> log/celeryd-w1.log <==
2018-08-29 17:02:36,050 ERROR    analysis_manager.tasks:280 _refinery_file_import() - Analysis 'FastQC - 2018/08/29 17:02:24 - Scott_Ouellette - scott_ouellette - Galaxy workflow execution for: FastQC' failed during file import

>>> from celery.result import AsyncResult
>>> from file_store.models import FileStoreItem
>>> f = FileStoreItem.objects.get(uuid="167cc543-99ff-4de0-b67c-bca791de594f")
>>> f.get_import_status()
'PENDING'
```

If you want to update the title of this issue or create a new one it doesn't matter to me, but both behaviors are due to that same portion of code that I referenced.

**Comment by hackdna** [comment]
Thanks for providing the details. I'll look more into this. In the mean time, do you have any suggestions for a fix?

**Comment by scottx611x** [comment]
Maybe we could add something like:
```
if not item.is_local():
```
before that chunk I referenced in `import_file()` to determine if a given file has already been successfully imported into Refinery, and skip the logic that is causing this issue.

We could also [jump up a level and not run an `import_file` task](https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/analysis_manager/tasks.py#L258-L259) if we know that a file has already been successfully imported.
